### PR TITLE
[const evaluator] Refactoring: exposing functions to look up symbolic values and to flow-sensitively evaluate possibly branching instructions.

### PR DESCRIPTION
**Pull request by ravikandhadai**

### CODE REVIEW

**Comment by ravikandhadai** [comment]
@marcrasi @devincoughlin  This is a minor refactoring that exposes a couple of utility functions. These methods will be useful for creating a flow-sensitive interpreter that will enable a client to step through a sequence of instructions.

**Comment by ravikandhadai** [comment]
@swift-ci Please smoke test

File: lib/SILOptimizer/Utils/ConstExpr.cpp [hunk 1]
```diff
@@ -1285,22 +1409,14 @@ static llvm::Optional<SymbolicValue> evaluateAndCacheCall(
...
 
-    // If we can evaluate this flow sensitively, then keep going.
-    if (!isa<TermInst>(inst)) {
-      auto fsResult = state.evaluateFlowSensitive(inst);
-      if (fsResult.hasValue())
-        return fsResult;
-      continue;
-    }
-
     // Otherwise, we handle terminators here.
```
Comment (marcrasi) [inline lib/SILOptimizer/Utils/ConstExpr.cpp]: "Otherwise" doesn't make sense any more

**Review by marcrasi — APPROVED** [review]

File: lib/SILOptimizer/Utils/ConstExpr.cpp (hunk 1 above)
Comment (ravikandhadai) [reply lib/SILOptimizer/Utils/ConstExpr.cpp]: Sure. Thanks for the review.

**Comment by ravikandhadai** [comment]
@swift-ci Please test and merge

**Comment by ravikandhadai** [comment]
@swift-ci Please test and merge

**Comment by ravikandhadai** [comment]
@swift-ci Please test Linux Platform

**Comment by ravikandhadai** [comment]
@swift-ci Please test Linux Platform

**Pull request merged**